server/datastore/mysql/hosts.go

```diff
@@ -51,17 +50,37 @@ func (d *Datastore) NewHost(ctx context.Context, host *fleet.Host) (*fleet.Host,
 		host.OSVersion,
 		host.Uptime,
 		host.Memory,
-		host.SeenTime,
 		host.TeamID,
 	)
 	if err != nil {
 		return nil, errors.Wrap(err, "new host")
 	}
 	id, _ := result.LastInsertId()
 	host.ID = uint(id)
+
+	_, err = d.writer.ExecContext(ctx, `INSERT INTO host_seen_times (host_id, seen_time) VALUES (?,?)`, host.ID, host.SeenTime)
+	if err != nil {
+		return nil, errors.Wrap(err, "new host seen time")
+	}
+
 	return host, nil
 }
 
+func (d *Datastore) SerialSaveHost(ctx context.Context, host *fleet.Host) error {
+	errCh := make(chan error, 1)
+	defer close(errCh)
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case d.writeCh <- itemToWrite{
+		ctx:   ctx,
+		errCh: errCh,
+		item:  host,
+	}:
+		return <-errCh
```

Comment:
@lucasmrod @mna after a convo with Lucas, he helped me see that I could remove the select from here, and defer close errCh above. Just mentioning explicitly to bring more eyes here.

Reply:
Last note on my end, given latest state of the code, we don't need a buffered `errCh` channel anymore (the creator will always read it after writing to `writeCh`, but!... maybe it makes sense in the scenario where there's high load to not block the serializing goroutine...). So am ok with leaving as-is :)

Reply:
Yes that works, it didn't work in the previous implementation as the context could be cancelled and cause the function to return between the write to writeCh and the read from errCh. Now, it is guaranteed not to return before the other goroutine writes to `errCh`, though note that it means the context can only unblock the call if it expires before the write to `writeCh`. After that, it can block indefinitely waiting for the result from `errCh`. This should be fine though, as the item's ctx is also used in the other goroutine to stop the database actions early if necessary.